tests/Utils/SchemaExtenderTest.php

```diff
@@ -2106,4 +2108,29 @@ public function testSupportsTypeConfigDecorator(): void
 
         self::assertSame(['data' => ['hello' => 'Hello World!', 'foo' => ['value' => 'bar']]], $result->toArray());
     }
+
+    /**
+     * Tests both custom inline and class scalar definitions.
+     *
+     * Ensures the correct instance is maintained before and after schema extension.
+     * Should probably be incorporated into testExtendsWithoutAlteringOriginalSchema().
+     */
+    public function testExtendsWithoutAlteringOriginalScalarTypes(): void
+    {
+        $extendedSchema = $this->extendTestSchema(/** @lang GraphQL */ '
+          extend type Foo {
+            bar: Bar
+          }
+        ');
+
+        $someScalar              = $this->testSchema->getType('SomeScalar');
+        $someScalarClass         = $this->testSchema->getType('SomeScalarClass');
+        $extendedSomeScalar      = $extendedSchema->getType('SomeScalar');
+        $extendedSomeScalarClass = $extendedSchema->getType('SomeScalarClass');
+
+        self::assertInstanceOf(CustomScalarType::class, $someScalar);
+        self::assertInstanceOf(SomeScalarClassType::class, $someScalarClass);
+        self::assertInstanceOf(CustomScalarType::class, $extendedSomeScalar);
+        self::assertInstanceOf(SomeScalarClassType::class, $extendedSomeScalarClass);
```

Comment:
This test pointed me in the right direction, but I do not think it is correct. The `SchemaExtender` consistently produces new types, which is why `$someScalar !== $extendedSomeScalar`.

What should happen though is that the behaviour of the type does not get erased. Thus, the test should ensure the method of the original scalar class is called when using `$extendedSomeScalarClass`.

Reply:
Makes sense, but should that be the case if the scalar type isn't actually extended in the new Schema?  In my case, I was just using the scalar type that had been declared in the original schema rather than extending it.

Reply:
Also, can scalars types actually be extended?  I don't think you can do much with scalars in GQL and I didn't think you can extend a native schema with another in PHP.

Reply:
Certainly possible, see http://spec.graphql.org/draft/#sec-Scalar-Extensions. It is useful for adding directives.

There is consistency in always producing a new type.